universal-application-tool-0.0.1/app/controllers/admin/ProgramAdminManagementController.java

```diff
@@ -86,8 +87,24 @@ public Result update(Http.Request request, long programId) {
           roleService.makeProgramAdmins(
               programId, ImmutableSet.copyOf(manageAdminForm.getAdminEmails()));
       Result result = redirect(routes.AdminProgramController.index());
-      if (maybeError.isPresent()) {
-        return result.flashing("error", maybeError.get().message());
+
+      if (maybeError.isPresent()){   
+        Optional<Program> program =
+            programRepository.lookupProgram(programId).toCompletableFuture().join();
+        if (program.isEmpty()) {
+          return notFound(String.format("Program with ID %s was not found", programId));
+        } else {
+          ImmutableList<String> programAdmins =
+              programRepository.getProgramAdministrators(programId).stream()
+                  .map(Account::getEmailAddress)
+                  .collect(toImmutableList());
+          return ok(
+              manageAdminsView.render(
+                  request,
+                  program.get().getProgramDefinition(),
+                  programAdmins,
+                  maybeError.get().message()));
```

Comment:
This return statement is changing the behavior of the page after there's an error.

If there's an error saving a program admin (the account doesn't exist or the account is already a global admin), the page will stay on the Manage Admins page and show the error. Previously, it would redirect to the "All programs" page and show the error. I think it makes more sense to show the error and **stay** on the manage admins page -- this makes it clearer to the global admin that there was an issue/something wasn't saved that they intended to save.

@Reviewers feel free to let me know if you have differing opinions on this.

Reply:
Nice work, definitely an improvement in behavior